File: internal/tui/components/help.go
package components

import (
	"strings"

	"github.com/charmbracelet/bubbles/key"
	"github.com/charmbracelet/lipgloss"
	"github.com/mailersend/mailersend-cli/internal/tui/theme"
)

var (
	helpOverlayStyle = lipgloss.NewStyle().
				Border(lipgloss.RoundedBorder()).
				BorderForeground(theme.Primary).
				Padding(1, 2).
				Background(theme.BgOverlay)

	helpTitleStyle = lipgloss.NewStyle().
			Bold(true).
			Foreground(theme.Primary).
			MarginBottom(1)

	helpKeyStyle = lipgloss.NewStyle().
			Foreground(theme.Key).
			Width(14)

	helpDescStyle = lipgloss.NewStyle().
			Foreground(theme.TextSub)

	helpSectionStyle = lipgloss.NewStyle().
				MarginTop(1).
				MarginBottom(0)
)

// Help is the help overlay component.
type Help struct {
	bindings [][]key.Binding
	visible  bool
	width    int
	height   int
}

// NewHelp creates a new help overlay.
func NewHelp(bindings [][]key.Binding) Help {
	return Help{
		bindings: bindings,
	}
}

// SetVisible sets whether the help overlay is shown.
func (h *Help) SetVisible(visible bool) {
	h.visible = visible
}

// Toggle toggles the help overlay visibility.
func (h *Help) Toggle() {
	h.visible = !h.visible
}

// Visible returns whether the help is showing.
func (h Help) Visible() bool {
	return h.visible
}

// SetSize sets the overlay dimensions.
func (h *Help) SetSize(width, height int) {
	h.width = width
	h.height = height
}

// View renders the help overlay.
func (h Help) View() string {
	if !h.visible {
		return ""
	}

	var b strings.Builder

	b.WriteString(helpTitleStyle.Render("Keyboard Shortcuts"))
	b.WriteString("\n\n")

	sections := []struct {
		title    string
		bindings []key.Binding
	}{
		{"Navigation", h.bindings[0]},
		{"Actions", h.bindings[1]},
		{"Views", h.bindings[2]},
		{"General", h.bindings[3]},
	}

	for i, section := range sections {
		if i > 0 {
			b.WriteString("\n")
		}
		b.WriteString(helpSectionStyle.Render(lipgloss.NewStyle().Bold(true).Render(section.title)))
		b.WriteString("\n")

		for _, binding := range section.bindings {
			help := binding.Help()
			line := helpKeyStyle.Render(help.Key) + helpDescStyle.Render(help.Desc)
			b.WriteString(line)
			b.WriteString("\n")
		}
	}

	b.WriteString("\n")
	b.WriteString(lipgloss.NewStyle().Foreground(theme.Muted).Render("Press ? or Esc to close"))

	content := b.String()

	// Calculate overlay size
	overlayWidth := 40
	overlayHeight := strings.Count(content, "\n") + 4

	// Center the overlay
	x := (h.width - overlayWidth) / 2
	y := (h.height - overlayHeight) / 2
	if x < 0 {
		x = 0
	}
	if y < 0 {
		y = 0
	}

	styled := helpOverlayStyle.Width(overlayWidth).Render(content)

	// Add positioning
	var result strings.Builder
	for i := 0; i < y; i++ {
		result.WriteString("\n")
	}
	lines := strings.Split(styled, "\n")
	for _, line := range lines {
		result.WriteString(strings.Repeat(" ", x))
		result.WriteString(line)
		result.WriteString("\n")
	}

	return result.String()
}

File: src/tui/components/help.rs
use ratatui::layout::Rect;
use ratatui::style::{Modifier, Style};
use ratatui::text::{Line, Span};
use ratatui::widgets::{Block, BorderType, Clear, Padding, Paragraph};
use ratatui::Frame;

use super::pad_right;
use crate::tui::keys;
use crate::tui::theme;

pub fn render(frame: &mut Frame, area: Rect) {
    let key_style = Style::new().fg(theme::KEY);
    let desc_style = Style::new().fg(theme::TEXT_SUB);

    let mut lines: Vec<Line<'static>> = vec![
        Line::styled(
            "Keyboard Shortcuts",
            Style::new().fg(theme::PRIMARY).add_modifier(Modifier::BOLD),
        ),
        Line::raw(""),
    ];

    for (i, section) in keys::HELP_SECTIONS.iter().enumerate() {
        if i > 0 {
            lines.push(Line::raw(""));
        }
        lines.push(Line::styled(
            section.title,
            Style::new().add_modifier(Modifier::BOLD),
        ));
        for binding in section.bindings {
            lines.push(Line::from(vec![
                Span::styled(pad_right(binding.key, 14), key_style),
                Span::styled(binding.desc, desc_style),
            ]));
        }
    }

    lines.push(Line::raw(""));
    lines.push(Line::styled(
        "Press ? or Esc to close",
        Style::new().fg(theme::MUTED),
    ));

    let overlay_width = 40u16.min(area.width);
    let overlay_height = (lines.len() as u16 + 4).min(area.height);
    let x = area.x + area.width.saturating_sub(overlay_width) / 2;
    let y = area.y + area.height.saturating_sub(overlay_height) / 2;
    let overlay = Rect::new(x, y, overlay_width, overlay_height);

    let block = Block::bordered()
        .border_type(BorderType::Rounded)
        .border_style(Style::new().fg(theme::PRIMARY))
        .padding(Padding::new(2, 2, 1, 1))
        .style(Style::new().bg(theme::BG_OVERLAY));

    frame.render_widget(Clear, overlay);
    frame.render_widget(Paragraph::new(lines).block(block), overlay);
}
